app/models/manageiq/providers/nuage/network_manager/cloud_subnet.rb

```diff
@@ -1,2 +1,60 @@
 class ManageIQ::Providers::Nuage::NetworkManager::CloudSubnet < ::CloudSubnet
+  supports :delete
+  supports :create
+
+  def delete_cloud_subnet_queue(userid)
+    task_opts = {
+      :action => "deleting Cloud Subnet for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'raw_delete_cloud_subnet',
+      :instance_id => id,
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :args        => []
+    }
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def raw_delete_cloud_subnet
+    $nuage_log.info("Deleting Cloud Subnet (ems_ref = #{ems_ref})")
+
+    Ansible::Runner.run(
+      ext_management_system.ansible_env_vars,
+      ext_management_system.ansible_extra_vars(:id => ems_ref),
+      ext_management_system.playbook('remove-subnet.yml')
+    )
+
+    $nuage_log.info("Done deleting Cloud Subnet (ems_ref = #{ems_ref})")
+  rescue => e
+    $nuage_log.error "Error deleting Cloud Subnet: #{e}"
+    raise MiqException::MiqCloudSubnetDeleteError
+  end
+
+  def self.raw_create_cloud_subnet(ext_management_system, options)
+    $nuage_log.info("Create Cloud Subnet (ems_ref = #{ems_ref})")
+
+    # TODO: update UI to match what Nuage needs
+
+    subnet = Ansible::Runner.run(
+      ext_management_system.ansible_env_vars,
+      ext_management_system.ansible_extra_vars(
+        :domain_id => options[:router_id],
+        :subnet_attributes => { :name => options[:name] }
+      ),
+      ext_management_system.playbook('remove-subnet.yml')
+    )
+
+    $nuage_log.info("Done creating Cloud Subnet (ems_ref = #{ems_ref})")
+
+    # TODO: can playbook return value?
+
+    {:ems_ref => subnet.id, :name => options[:name]}
```

Comment:
@Ladas @agrare so here we'd need playbook to return some values. We have two options:

- don't parse playbook output at all
- parse what playbook is printing to the STDOUT

Which approach would you think is better?

Reply:
I think returning the ref and name is a good thing in case the caller wants to do a state-machine loop waiting for it to show up in our inventory.  If this is available in the playbook we should print it and parse the stdout

Reply:
I will prepare for you the parsed_stdout, where you should be able to find it.

E.g. in my example, I use ```set_stats``` module to store result:

https://github.com/ManageIQ/manageiq-providers-amazon/pull/453/files#diff-5f61cf6a1eccbf28f5617e02b272fdd1R11

Then doing ```pp parsed_stdout.last["event_data"]["artifact_data"]```, should give all the set_stat records, like this:

```ruby
{"created_security_group"=>
  {"ip_permissions"=>
    [{"from_port"=>22,
      "ip_protocol"=>"tcp",
      "to_port"=>22,
      "ip_ranges"=>[{"cidr_ip"=>"0.0.0.0/0"}],
      "prefix_list_ids"=>[],
      "ipv6_ranges"=>[],
      "user_id_group_pairs"=>[]}],
   "description"=>"ladas_test_changed",
   "tags"=>{},
   "changed"=>false,
   "group_name"=>"ladas_test5",
   "failed"=>false,
   "vpc_id"=>"vpc-ff49ff91",
   "group_id"=>"sg-14ca085e",
   "ip_permissions_egress"=>
    [{"user_id_group_pairs"=>[],
      "prefix_list_ids"=>[],
      "ip_ranges"=>[{"cidr_ip"=>"0.0.0.0/0"}],
      "ip_protocol"=>"-1",
      "ipv6_ranges"=>[]}],
   "owner_id"=>"200278856672"}}
```

Reply:
note: this is returned also if we update the security group, so we should also check if the particular play was changed. Or maybe it doesn't matter that much

Reply:
also set_stats is cumulative, so calling it multiple times with the same var, should result in having array with all the values

Reply:
As I said, AnsibleRunner is a love at first sight 😍 Thanks, what you describe sounds like really cool and managable, can't wait ansible-runner PR being merged.